homeassistant/components/media_player/__init__.py

```diff
@@ -674,6 +675,31 @@ def state_attributes(self):
 
         return state_attr
 
+    def preload_media_image_url(self, image_url):
+        """Preload and cache a media image for future use."""
+        self.hass.loop.call_later(
+            0.1,
+            _fetch_image,
+            image_url
+        )
+
+
+@functools.lru_cache(maxsize=16)
+def _fetch_image(url):
+    """Helper method to fetch image.
+
+    Up to 16 images are cached in memory (the images are typically 10-100kB in
+    size).
+    """
+    try:
+        response = requests.get(url)
```

Comment:
I get where you're coming from :) Yes this it will be called from within the event loop, and if I change it to use aiohttp it will need to be a coroutine and that it would stop working, so you're right :). Will have a look at another approach.